docs/how-to-guides/data-basics/5-adding-a-delete-button.md

```diff
@@ -0,0 +1,437 @@
+	const { createErrorNotice } = useDispatch( noticesStore );
+	// ...
+	const { error, isDeleting } = useSelect(
+		select => ( {
+			error: select( coreDataStore ).getLastEntityDeleteError( 'postType', 'page', pageId ),
+			isDeleting: select( coreDataStore ).isDeletingEntityRecord( 'postType', 'page', pageId ),
+		} ),
+		[ pageId ]
+	);
+
+	useEffect( () => {
+		if ( error ) {
+			const message = ( error?.message || 'There was an error.' ) + ' Please refresh the page and try again.'
+			createErrorNotice( , {
+				type: 'snackbar',
+			} );
+		}
+	}, [ error ] )
+    // ...
+}
+```
+
+Great! `DeletePageButton` is now fully aware of errors. Let's see that error message in action. We'll trigger an invalid delete and let it fail. One way to do it is to set the `id` to `-1`:
+
+```js
+export function DeletePageButton( { pageId, onCancel, onSaveFinished } ) {
+	// ...
+	pageId = pageId * -1;
+	const handleDelete = () => deleteEntityRecord( 'postType', 'page', pageId );
+	// ...
+}
+```
+
+Once you refresh the page and click any `Delete` button, you should see the following error message:
+
+![](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/how-to-guides/data-basics/media/delete-button/snackbar-error.png)
+![](./media/delete-button/snackbar-error.png)
+
+Fantastic! We can now **remove the `pageId = pageId * -1;` line.**
+
+Before we move on to the next step, let's also confirm a successful deletion with another snackbar message:
+
+```js
+
+function DeletePageButton( { pageId } ) {
+	const { deleteEntityRecord } = useDispatch( coreDataStore );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const handleDelete = async () => {
+		const success = await deleteEntityRecord( 'postType', 'page', pageId );
+		if ( success ) {
+			createSuccessNotice( "The page was deleted!", {
+				type: 'snackbar',
+			} );
+		}
+	}
+	// ...
+}
+```
+
+Here's what you should see after applying these changes and deleting a page:
+
+![](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/how-to-guides/data-basics/media/delete-button/snackbar-success.png)
+![](./media/delete-button/snackbar-success.png)
+
+And that's it!
+
+### Wiring it all together
+
+All the pieces are in place, great! Here’s all the changes we've made in this chapter:
+
+```js
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { Button, Modal, TextControl } from '@wordpress/components';
+
+function MyFirstApp() {
+	const [searchTerm, setSearchTerm] = useState( '' );
+	const { pages, hasResolved } = useSelect(
+		( select ) => {
+			const query = {};
+			if ( searchTerm ) {
+				query.search = searchTerm;
+			}
+			const selectorArgs = ['postType', 'page', query];
+			const pages = (
+				select( coreDataStore )
+					.getEntityRecords( ...selectorArgs )
+					?.filter( page => ! select( coreDataStore ).isDeletingEntityRecord('postType', 'page', page.id ) )
```

Comment:
This is making sure we're not displayed any pages currently being deleted right?